contrib/storage-http/src/test/java/org/apache/drill/exec/store/http/TestURLParameters.java

```diff
@@ -0,0 +1,121 @@
+    filters3.put("param1", "value1");
+    filters3.put("repos", "drill");
+    assertEquals("https://github.com/orgs/org/repos", SimpleHttp.mapURLParameters(githubNoParam, filters3));
+  }
+
+  @Test
+  public void testParamAtEnd() {
+    HttpUrl pokemonUrl = HttpUrl.parse("https://pokeapi.co/api/v2/pokemon/{pokemon_name}");
+    CaseInsensitiveMap<String> filters = CaseInsensitiveMap.newHashMap();
+    filters.put("pokemon_name", "Misty");
+    filters.put("param1", "value1");
+    filters.put("repos", "drill");
+    assertEquals("https://pokeapi.co/api/v2/pokemon/Misty", SimpleHttp.mapURLParameters(pokemonUrl, filters));
+  }
+
+  @Test
+  public void testUpperCase() {
+    HttpUrl githubSingleParam = HttpUrl.parse("https://github.com/orgs/{ORG}/repos");
+    CaseInsensitiveMap<String> filters = CaseInsensitiveMap.newHashMap();
+    filters.put("org", "apache");
+    filters.put("param1", "value1");
+    filters.put("param2", "value2");
+    assertEquals("https://github.com/orgs/apache/repos", SimpleHttp.mapURLParameters(githubSingleParam, filters));
+  }
+
+  @Test
+  public void testURLDefaultParameters() {
+    HttpUrl githubSingleParam = HttpUrl.parse("https://github.com/orgs/{org=apache}/repos");
+    CaseInsensitiveMap<String> filters = CaseInsensitiveMap.newHashMap();
+    filters.put("param1", "value1");
+    filters.put("param2", "value2");
+    assertEquals("https://github.com/orgs/apache/repos", SimpleHttp.mapURLParameters(githubSingleParam, filters));
+  }
+
+  @Test
+  public void testMixedCase() {
+    // Since SQL is case-insensitive,
+    HttpUrl githubSingleParam = HttpUrl.parse("https://github.com/orgs/{ORG}/{org}/repos");
+    CaseInsensitiveMap<String> filters = CaseInsensitiveMap.newHashMap();
+    filters.put("org", "apache");
+    filters.put("ORG", "linux");
+    filters.put("param1", "value1");
+    filters.put("param2", "value2");
+    assertEquals("https://github.com/orgs/linux/linux/repos", SimpleHttp.mapURLParameters(githubSingleParam, filters));
+  }
+
+  @Test
+  public void testDuplicateParameters() {
+    HttpUrl pokemonUrl = HttpUrl.parse("https://pokeapi.co/api/{pokemon_name}/pokemon/{pokemon_name}");
+    CaseInsensitiveMap<String> filters = CaseInsensitiveMap.newHashMap();
+    filters.put("pokemon_name", "Misty");
+    filters.put("param1", "value1");
+    filters.put("repos", "drill");
+    assertEquals("https://pokeapi.co/api/Misty/pokemon/Misty", SimpleHttp.mapURLParameters(pokemonUrl, filters));
+  }
+
+  @Test
+  public void testDefaultParametersWithDifferentDatatypes() {
+    HttpUrl pokemonUrl = HttpUrl.parse("https://pokeapi.co/api/{boolean=true}/{int=1234}");
+    CaseInsensitiveMap<String> filters = CaseInsensitiveMap.newHashMap();
+    assertEquals("https://pokeapi.co/api/true/1234", SimpleHttp.mapURLParameters(pokemonUrl,filters));
+  }
+
+  @Test
+  public void testDefaultParameterExtractor() {
+    HttpUrl pokemonUrl = HttpUrl.parse("https://pokeapi.co/api/{pokemon_name=Misty}");
```

Comment:
Would be great to test the `{foo=}` case as well to define the expected behavior.

Reply:
@paul-rogers 
I added a unit test for this.  Upon reflecting on it a bit, it would seem to me that the "correct" behavior would be to throw an exception in this case.  Having a blank default rather defeats the purpose of having a default at all, so my thinking here, was either specify a default value or don't.

Does that make sense?   In any event, I think I've addressed all your comments.  Thanks for the review!